haproxy.cfg

```diff
@@ -27,12 +27,14 @@ frontend http_front
     mode http
     log global
 
+    redirect scheme https code 301 if !{ ssl_fc }
```

Comment:
Our port 80 operation got a ton simpler. Now if a request comes in on 80, we redirect the user to port 443 with the same path. It's transparent, you never even see it happen, but and we know that if we're serving it, we're serving it out under lock and key.